src/NServiceBus.Core/Routing/SubscriptionMigrationMode/SubscriptionMigrationModeSettings.cs

```diff
@@ -17,6 +17,7 @@ public class SubscriptionMigrationModeSettings : ExposeSettings
         /// </summary>
         public SubscriptionMigrationModeSettings(SettingsHolder settings) : base(settings)
         {
+            settings.Set("NServiceBus.Subscriptions.EnableMigrationMode", true);
```

Comment:
hm not sure I like this approach to introduce side effects when calling the constructor.

Reply:
The current way of enabled migration mode is super obscure and I'm pretty sure we have other examples in core where we already use or used this

Reply:
> The current way of enabled migration mode is super obscure 

I don't disagree with that statement. When it was added, it was considered to be "good enough" for the few and temporary use cases to not bother making this a more explicit API.

>  I'm pretty sure we have other examples in core where we already use or used this

I don't think this is a strong argument to justify violating this.
According to a quick look at core, the only settings type violating this rule is `PersistenceExtensions`.

Reply:
it's also true that we agreed upon making all public configuration explicit and not relying on settings and magic strings.

Reply:
> it's also true that we agreed upon making all public configuration explicit and not relying on settings and magic strings.

absolutely right @mauroservienti, not sure this applies here though as this is a "hidden feature flag", so there was an explicit decision to not making this discoverable.

Reply:
I agree with @timbussmann, not sure I like this. These sort of things are intended to be something the transport has control over without exposing them as a general configuration API.

Another example of this is the timeout manager migration mode setting.